test/Altinn.Platform.Authentication.Tests/Controllers/SystemUserControllerTest.cs

```diff
@@ -550,6 +586,68 @@ public async Task SystemUser_ListByVendorsSystem_ReturnsOk()
+            HttpResponseMessage response = await CreateSystemRegister(dataFileName);
+
+            HttpClient client = CreateClient();
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", PrincipalUtil.GetToken(1337, null, 3));
+
+            int partyId = 500000;
+  
+            SystemUserRequestDto newSystemUser = new()
+            {
+                IntegrationTitle = "IntegrationTitleValue",
+                SystemId = "991825827_the_matrix",
+            };
+
+            HttpRequestMessage createSystemUserRequest = new(HttpMethod.Post, $"/authentication/api/v1/systemuser/{partyId}/create")
+            {
+                Content = JsonContent.Create<SystemUserRequestDto>(newSystemUser, new MediaTypeHeaderValue("application/json"))
+            };
+
+            HttpResponseMessage createSystemUserResponse = await client.SendAsync(createSystemUserRequest, HttpCompletionOption.ResponseContentRead);
+
+            var result = await createSystemUserResponse.Content.ReadFromJsonAsync<CreateSystemUserResponse>();
+            Assert.Equal(HttpStatusCode.OK, createSystemUserResponse.StatusCode);
+            SystemUser? shouldBeCreated = result?.SystemUser;
+            Assert.NotNull(shouldBeCreated);
+            Assert.Equal(newSystemUser.SystemId, shouldBeCreated.SystemId);
+            Assert.Equal(newSystemUser.IntegrationTitle, shouldBeCreated.IntegrationTitle);
+        }
+
+        [Fact]
+        public async Task SystemUser_CreateAndDelegate_Returns_Error()
+        {
+            // Create System used for test
+            string dataFileName = "Data/SystemRegister/Json/SystemRegister.json";
+            HttpResponseMessage response = await CreateSystemRegister(dataFileName);
+
+            HttpClient client = CreateClient();
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", PrincipalUtil.GetToken(1337, null, 3));
+
+            int partyId = 500001;
+
+            SystemUserRequestDto newSystemUser = new()
+            {
+                IntegrationTitle = "IntegrationTitleValue",
+                SystemId = "991825827_the_matrix",
+            };
+
+            HttpRequestMessage createSystemUserRequest = new(HttpMethod.Post, $"/authentication/api/v1/systemuser/{partyId}/create")
+            {
+                Content = JsonContent.Create<SystemUserRequestDto>(newSystemUser, new MediaTypeHeaderValue("application/json"))
+            };
+
+            HttpResponseMessage createSystemUserResponse = await client.SendAsync(createSystemUserRequest, HttpCompletionOption.ResponseContentRead);
+            Assert.Equal(HttpStatusCode.BadRequest, createSystemUserResponse.StatusCode);  
+            var problemDetails = await createSystemUserResponse.Content.ReadFromJsonAsync<ProblemDetails>();
+            Assert.Equal(Problem.Reportee_Orgno_NotFound.Detail, problemDetails.Detail);
```

Comment:
Er det ikke mulig å lage noen positive tester?

Reply:
Det er to positive tester direkte på endepunktet. Og flere eksisterende tester bekrefter at refactoring ikke ødela funksjonalitet for CreateRequest, CreateChangeRequest, Approve Request og Approve ChangeRequest. Det må AT testing til for å se om det spiller sammen med BFF/UI.

Reply:
Lagt til tre xunit tester til